packages/grafana-ui/src/components/IconButton/IconButton.tsx

```diff
@@ -36,7 +36,7 @@ export const IconButton = React.forwardRef<HTMLButtonElement, Props>(
     const styles = getStyles(theme, size, variant);
 
     const button = (
-      <button ref={ref} {...restProps} className={cx(styles.button, className)}>
+      <button ref={ref} aria-label={tooltip} {...restProps} className={cx(styles.button, className)}>
```

Comment:
Not sure this is the best solution, most icon buttons don't have tooltip, so they still won't be accessible, I think a better way would be to add `aria-label` prop to the component, so the consumers could provide the labels according to their use case.

Reply:
What about taking the tooltip if there is no aria-label? That way we cover a lot of cases that already exist. You can always override it through restProps.

Reply:
I'm thinking that tooltip doesn't always correctly represent the button, content. For example when it's some help text, also sometimes tooltips are an html content (not in this case but might happen in the future). Could be an edge case though.

Reply:
I don't think it fits the use case for an icon button. Icon buttons are minimalist and should fit a very specific action. Maybe we can add guidelines about it.

Reply:
Not sure if this is the *BEST* approach, but using https://benjaminjohnson.me/blog/accessible-icon-buttons as a reference, probably adding a required `label` prop, adding its content next to the icon, and "hiding" the text to non-screenreader users via css might be a good approach. wdyt?

big plus imho: an accessible description of the "action" will be required

Reply:
Sounds good but the issue is that this will break all the current use-cases, so I think we'd need to come up with a backwards compatible approach for now and wait with the breaking changes until the next major release.

Reply:
@Elfo404 my biggest problem with the hide-from-screenreader approach is how crazy that css has to be (as the linked article describes, you cannot just visibility=hidden or display=none). if there would be a cleaner way to mark it as screen-reader only :-( . though i understand it can be written once and then reused everywhere.

Reply:
maybe it would be interesting to go through a couple well known websites and see how they deal with icon-only buttons, it seems to me the web is full of them :-) 
i did a quick check on gmail/google-sheets, and they went with `aria-label`, the same for the github formatting tools right above the text i am writing right now.

Reply:
@Clarity-89 true, then i see no problem in having it optional. maybe we can log a warning in the console before 9.0 if the props isn't used? not sure if we can add a linter rule for that.

Reply:
@gabor that's interesting, somehow i always thought using aria-label wasn't the proper way to solve this specific use case, but reading more about the subject it seems it actually is

Reply:
@Elfo404 yeah an optional property with a warning sounds good imo.

Reply:
I think the big issue here is that our tooltips are not WCAG compliant, otherwise they would've served this purpose.